### PR TITLE
feat(ux): clearer agent-facing contracts (closes #77, #81)

### LINKED ISSUE #77: Agent UX papercuts: argument validation messages, project_health scope, engine labels in tool descriptions
## Goal

Three small, independent UX fixes that surfaced during a real agent debugging session today. All three are papercuts — none break workflows, but each costs 1–2 confusing tool calls per session for an agent until it learns the quirk.

Filed as one issue because all three are <30 LOC each and share a theme (clearer agent-facing contracts).

---

### What happened
I wanted to "find every use of `JsonDocument` across `src/FsMcp.Server`". Naturally I tried:

```jsonc
{ "path": "/Users/roman/dev/FsMcp/src/FsMcp.Server",
  "projectPath": ".../FsMcp.Server.fsproj",
  "symbolQuery": "JsonDocument" }
```

Got:

```
TransportError "{\"errorKind\":\"InfraFailure\",\"message\":\"Access to the path '/Users/roman/dev/FsMcp/src/FsMcp.Server' is denied.\"}"
```

The error message led me to investigate file permissions for two minutes before I realised the contract is "single .fs file as anchor, scope comes from `projectPath`".

### Suggested fix
Validate `path` early; if it's a directory, return:

```jsonc
{ "status": "error",
  "errorKind": "InvalidArgument",
  "message": "fcs_find_symbol expects 'path' to be a source file (.fs/.fsi), not a directory. To search project-wide, pass any source file in the project as 'path' and the .fsproj as 'projectPath'." }
```

Same treatment for any other tool with a `path` arg that semantically means "anchor file".

---

### What happened
First call after `set_project` (or in a fresh session before any tool that needs LSP):

```jsonc
{ "status": "ok",
  "reportKind": "project",
  "toolingReadiness": {
    "status": "degraded",
    "warnings": ["LSP workspace is not initialized or not ready."]
  },
  ...
}
```

If I'm planning to use only `fcs_*` and `fsharp_compile` (FCS in-process), an LSP not being warm yet is **irrelevant** — but `degraded` reads as "something is off". Mild trust erosion.

### Suggested fix
Split readiness into independent axes:

```jsonc
"toolingReadiness": {
  "fcs":       { "status": "ready" },
  "lsp":       { "status": "not_ready", "reason": "workspace not initialized; auto-warmed on first textDocument_*/workspace_* call" },
  "overall":   "fcs_only"
}
```

`overall = "fcs_only"` is a positive value meaning "everything an FCS-only agent needs is fine". Reserve `degraded` for cases where the requested-or-likely path is actually broken.

---

### What happened
This isn't documented end-to-end, so I had to infer it: `fcs_*` and `fsharp_compile` go through `FSharp.Compiler.Service` in-process; `textDocument_*` and `workspace_*` go through child `fsautocomplete`. That difference matters because:

- FCS in-process is fast and doesn't need LSP warm-up; `set_project` is optional.
- FSAC is heavier, requires `set_project`, and adds a separate process (with its own memory footprint — see #18).

### Suggested fix
Add a one-line `engine: "fcs" | "fsac"` (or a leading `[FCS]` / `[FSAC]` tag) to every tool's `description` field returned in `tools/list`. Optional secondary: surface this on the [README "What You Get" section](https://github.com/Neftedollar/FsLangMCP#what-you-get) so humans see it too.

Concretely, the `tools/list` envelope an agent sees would change from:

```json
{ "name": "fcs_find_symbol",
  "description": "Agent-friendly project-wide symbol search ..." }
```

to:

```json
{ "name": "fcs_find_symbol",
  "description": "[FCS in-process] Agent-friendly project-wide symbol search ..." }
```

This pre-empts an entire category of "do I need to call set_project first?" agent confusion.

---

## Non-goals

- Don't redesign error envelopes globally — these three changes can each be a 5–20 LOC patch in their respective sites.
- Don't introduce a new `engine` enum surface for callers (this is purely descriptive in tool metadata).

## Acceptance criteria

- [ ] `fcs_find_symbol` returns a structured `InvalidArgument` error with a clear message when `path` points at a directory (and ideally for any non-existent / non-readable file too).
- [ ] `project_health` decouples FCS-readiness from LSP-readiness and avoids `degraded` when the FCS axis alone is sufficient.
- [ ] Every MCP tool's `description` indicates its engine (FCS in-process vs child fsautocomplete) so agents can plan calls without a `set_project` wait.

### LINKED ISSUE #81: ProjectHealth.createReport early-exit paths still emit flat toolingReadiness shape
## Problem

In #77, `project_health` was restructured so `toolingReadiness` is split into independent axes:

```jsonc
"toolingReadiness": {
  "fcs":     { "status": "ready" },
  "lsp":     { "status": "not_ready", "reason": "..." },
  "overall": "fcs_only"
}
```

But the **early-exit paths** in `ProjectHealth.createReport` (when `resolveHealthProjectPath` fails, or when `tryReadProject` returns an error) were not updated. They still emit the legacy flat shape:

```jsonc
"toolingReadiness": { "status": "blocked", ... }
```

This means an agent that calls `project_health` on a missing project or unreadable .fsproj gets a structurally different `toolingReadiness` from the happy-path response, which breaks any client doing structural parsing.

## Expected

All exit paths emit the new nested shape:

```jsonc
"toolingReadiness": {
  "fcs":     { "status": "blocked", "reason": "<why>" },
  "lsp":     { "status": "blocked", "reason": "<why>" },
  "overall": "blocked"
}
```

Or whatever the equivalent error-shape is — the point is the structure must match the happy path.

## Discovery

Found during the #77 BUILD pass; the implementing agent flagged it explicitly. Small cleanup, blocked nothing in #77 itself, but worth fixing soon to keep the contract uniform.

## Reference

- #77 — feature that introduced the new shape

### PR DESCRIPTION
**Pull request by Neftedollar**
## Summary

Three small, independent agent-UX improvements filed together because each is <100 LOC and shares a "clearer agent-facing contracts" theme. Closes #77 and rolls in #81 (a related shape inconsistency surfaced during review).

- **A — `fcs_find_symbol` argument validation**: when `path` is a directory or unreadable, returns a structured `InvalidArgument` with a clear message instead of the cryptic `"Access … denied"`. Same treatment applied at all 8 FCS tool sites that take a `path` arg meaning "anchor file". When the caller supplies non-empty `text`, the `File.Exists` gate is bypassed (preserves the unsaved-buffer workflow); the `Directory.Exists` rejection is kept.
- **B — `project_health` axis split**: `toolingReadiness` is restructured into `{fcs, lsp, overall}`. `overall = "fcs_only"` is a *positive* value meaning "everything an FCS-only agent needs is fine"; `degraded` is reserved for cases where FCS itself is broken. Both happy-path and early-exit paths emit the new shape (closes #81).
- **C — Engine labels in tool descriptions**: every tool's `description` is prefixed with `[FCS in-process]` or `[FSAC]` so agents can plan calls without round-tripping a `set_project` warm-up.

## Verification

- `dotnet build`: 0 warnings, 0 errors.
- `dotnet test`: 73 passed (61 baseline + 12 new across all three parts).
- Two independent reviews + one closure re-verification:
  - **Code reviewer (initial)**: NEEDS WORK — flagged the unsaved-buffer regression and the `ProjectHealth` shape mismatch.
  - **RealityChecker (initial)**: OK WITH CAVEATS — flagged LSP-side path validation gap (out of scope per issue body) and symlink edge case.
  - **RealityChecker (closure)**: OK — all prior findings closed by named tests at FcsBridgeTests.fs:680/700/723 and ProjectHealthTests.fs:236-251/254/277/301.

## Files changed

| File | Change |
|---|---|
| `FcsBridge.fs` | `validateSourcePath text` helper applied at 8 FCS tool sites; bypasses `File.Exists` when non-empty `text`, keeps `Directory.Exists` rejection |
| `ProjectHealth.fs` | `toolingReadiness` split into `fcs`/`lsp`/`overall` axes on **all** exit paths (success + 2 early-exits); corrected `overallStatus` match table |
| `Program.fs` | All 21 tool descriptions prefixed `[FCS in-process]` (fcs_*, fsharp_compile, project_health, fsharp_project_inspect, fcs_get_project_options) or `[FSAC]` (textDocument_*, workspace_*, set_project, fsharp_signature_data) |
| `tests/FcsBridgeTests.fs` | +7 tests: directory rejection (3 sites), unsaved-buffer bypass (2 sites), missing-file rejection (2 sites) |
| `tests/ProjectHealthTests.fs` | +5 new + 2 updated: nested-shape happy path, `fcs_only`/`ready` overall, early-exit nested shape, blocked-fcs propagation |

## Test plan

- [x] `dotnet build` clean
- [x] `dotnet test` — 73/73 passing
- [x] Code review (initial NEEDS WORK → closed by fix commit)
- [x] Reality check (initial caveats + closure OK)
- [ ] Manual smoke: agent calls `fcs_find_symbol` with a directory path → sees clear InvalidArgument; agent calls `project_health` on an FCS-only project → sees `overall: "fcs_only"` not `"degraded"`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: FcsBridge.fs [hunk 1]
```diff
@@ -290,6 +290,35 @@ type internal FcsBridge() =
...
+                jobj
+                    [ "status", jstr "error"
+                      "errorKind", jstr "InvalidArgument"
+                      "message",
+                      jstr
+                          $"%s{toolName} expects 'path' to be a source file (.fs/.fsi), not a directory. To search project-wide, pass any source file in the project as 'path' and the .fsproj as 'projectPath'." ]
+                :> JsonNode
+            )
+        elif not hasText && not (File.Exists(fullPath)) then
+            Some(
```
[the remark this answers is not in the document]
Comment (Neftedollar) [reply FcsBridge.fs:312]: Already fixed in main: `validateSourcePath` treats any `Some _` as a provided buffer (`text |> Option.isSome`), so `text = ""` no longer forces `File.Exists` on a synthetic path. The code comment records the reason — an empty unsaved file is a valid editor state.
